pages/poll/[id]/results.tsx

```diff
@@ -0,0 +1,105 @@
+import type { NextPage } from 'next';
+import { useRouter } from 'next/router';
+import React, { useEffect, useState } from 'react';
+import Typography from '@mui/material/Typography';
+import Button from '@mui/material/Button';
+import SaveIcon from '@mui/icons-material/Save';
+import CircularProgress from '@mui/material/CircularProgress';
+import Page from '../../../components/Page';
+import styles from './results.module.css';
+import { GetPollResultsRequest } from '../../../components/api';
+import { useFetch } from '../../../hooks/useFetch';
+import YourVote from '../../../components/YourVote';
+import StepVisualization from '../../../components/StepVisualization';
+import VoteLine from '../../../components/VoteLine';
+
+const PollResultsPage: NextPage = () => {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const [pollResultsData, getPollResultsData] = useFetch(GetPollResultsRequest);
+  const [alert, setAlert] = useState<AlertShape | null>(null);
+  const [currentStep, setCurrentStep] = useState(1);
+
+  useEffect(() => {
+    // On the initial Page Load, this ID comes up as undefined
+    // but then gets populated with the actual ID from the URL
+    if (!id) {
+      return;
+    }
+
+    getPollResultsData(String(id));
+  }, [id]);
+
+  useEffect(() => {
+    // We don't want to spawn any alerts from before the request is sent. This is pretty janky
+    if (pollResultsData.isInitial) {
+      return;
+    }
+    // If fetching the Poll is done loading and was a failure then throw an alert message to the user
+    if (!pollResultsData.isLoading && !pollResultsData.isSuccess) {
+      setAlert({ severity: 'error', message: 'An error has occured while loading the poll results' });
+    }
+
+    // Clear the alert when loading or on success
+    if (pollResultsData.isLoading || pollResultsData.isSuccess) {
+      setAlert(null);
+    }
+  }, [pollResultsData.isInitial, pollResultsData.isLoading, pollResultsData.isSuccess]);
+
+  const goBackStep = () => {
+    const nextStep = Math.max(1, currentStep - 1);
+    setCurrentStep(nextStep);
+  };
+
+  const goForwardStep = () => {
+    const nextStep = Math.min(pollResultsData.data.steps.length, currentStep + 1);
+    setCurrentStep(nextStep);
+  };
+
+  const onFirstStep = currentStep <= 1;
+  const onLastStep = currentStep >= pollResultsData?.data?.steps?.length;
```

Comment:
Don't these need to be in a useEffect?
Maybe it doesn't matter because the component re-runs when currentStep changes?

Reply:
It doesn't need to be since it will just re-calculate when the props change